unstructured/metrics/text_extraction.py

```diff
@@ -50,3 +52,49 @@ def calculate_edit_distance(
     elif return_as == "distance":
         return distance
     return 0.0
+
+
+# Duplicate code from cleaners.core,
+# not sure we want this functionality introduced in the main library.
+def remove_punctuation(s: str, exclude_punctuation: Optional[list]) -> str:
+    """Removes punctuation from a given string."""
+
+    tbl = dict.fromkeys(
+        i for i in range(sys.maxunicode) if unicodedata.category(chr(i)).startswith("P")
+    )
+
+    if exclude_punctuation:
```

Comment:
We should exclude punctuation not just by character but by functionality - for example, remove dashes but not hyphens. maybe use the P_ [unicode categories](https://pypi.org/project/unicategories/) to determine which type of punctuation it is

Reply:
Not sure this does what we want. I think it just has a string of all the different types of hypens. It doesn't differentiate between where the hyphen is in a sentence.

Reply:
Hm. If there isn't already a way to do it using unicode categories, we can use regex or any other pattern matcher.
We want to match a "dash" with no spaces surrounding it (ex. easy-peasy) as being a hyphen that we want to keep in our output words, and everything else is a dash that is between separate words. I added a test for dash / hyphen

Reply:
Gotcha! I just used our `clean_bullet` brick to remove dashes from the beginning of the sentence and leave the hyphens. Is there another use case that I'm forgetting?